riscv-sbi.adoc

```diff
@@ -70,13 +70,44 @@ Standard SBI error codes are listed below
 
 [cols="<,>",options="header,compact"]
 |===
-|  Error Type              |Value
-|  SBI_SUCCESS             |  0
-|  SBI_ERR_FAILURE         | -1
-|  SBI_ERR_NOT_SUPPORTED   | -2
-|  SBI_ERR_INVALID_PARAM   | -3
-|  SBI_ERR_DENIED          | -4
-|  SBI_ERR_INVALID_ADDRESS | -5
+|  Error Type                |Value
+|  SBI_SUCCESS               |  0
+|  SBI_ERR_FAILED            | -1
+|  SBI_ERR_NOT_SUPPORTED     | -2
+|  SBI_ERR_INVALID_PARAM     | -3
+|  SBI_ERR_DENIED            | -4
+|  SBI_ERR_INVALID_ADDRESS   | -5
+|  SBI_ERR_ALREADY_AVAILABLE | -6
+|===
+
+Every SBI function should prefer `unsigned long` as the data type. It keeps the
+specification simple and easily adaptable for all RISC-V ISA types (i.e. RV32,
+RV64 and RV128). In case the data is defined as 32bit wide, the upper 32 bits
+should be considered as reserved bits unless otherwise specified.
+
+If a SBI function needs to pass a list of harts to the higher privilege mode, it
+must use a hart mask as defined below. This is applicable to any extensions defined
+in or after v0.2.
+
+Any function, requiring a hart mask, need to pass following two arguments.
+
+* `unsigned long hart_mask` is a scalar bit-vector containing hartids
+* `unsigned long hart_mask_base` is the starting hartid from which bit-vector
+   must be computed.
+
+In a single SBI function call, maximum number harts that can be set is always XLEN.
+If a lower privilege mode needs to pass information about more than XLEN harts, it
+should invoke multiple instances of the SBI function call.
```

Comment:
We should allow something to say "all harts", maybe -1 in _base?

Reply:
sure. Currently, I am manually modifying the hart mask to online mask if hart_mask is zero.
But assigning hbase to -1 will do as well. It will make life simple in OpenSBI as well :).